src/components/DataSession/DataSession.vue

```diff
@@ -15,52 +15,107 @@ const props = defineProps({
 
 const store = useStore()
 const emit = defineEmits(['reloadSession'])
+const images = ref([])
+const filteredImages = ref([])
 
 const dataSessionsUrl = store.state.datalabApiBaseUrl + 'datasessions/'
 const imagesPerRow = 4
+const operationMap = {}
+var selectedOperation = -1
+
+async function addCompletedOperationsOutput() {
+  const url = dataSessionsUrl + props.data.id + '/operations/'
+
+  function updateCompletedOperations(response) {
+    response.results.forEach(operationResponse => {
+      if (operationResponse.status == 'COMPLETED') {
+        addCompletedOperation(operationResponse)
+      }
+    })
+  }
+
+  await fetchApiCall({ url: url, method: 'GET', successCallback: updateCompletedOperations, failCallback: handleError })
+}
+
+
+function imagesContainsFile(file) {
+  return images.value.some(image => image.basename == file.basename && image.source == file.source && image.operation == file.operation)
+}
+
+
+function addCompletedOperation(operationResponse) {
+  if ('output' in operationResponse && 'output_files' in operationResponse.output) {
+    operationResponse.output.output_files.forEach(outputFile => {
+      outputFile.operation = operationResponse.id
+      outputFile.operationIndex = operationMap[operationResponse.id]
+      if (!imagesContainsFile(outputFile)) {
+        images.value.push(outputFile)
+        if (selectedOperation == -1 || selectedOperation == outputFile.operationIndex) {
+          filteredImages.value.push(outputFile)
+        }
+      }
+    })
+  }
+}
+
+
+function selectOperation(operationIndex) {
+  if (operationIndex != selectedOperation) {
+    selectedOperation = operationIndex
+    if (selectedOperation == -1) {
+      filteredImages.value = images.value
+    }
+    else {
+      filteredImages.value = images.value.filter(image => image.operationIndex == operationIndex)
+    }
+  }
+}
 
 async function addOperation(operationDefinition) {
   const url = dataSessionsUrl + props.data.id + '/operations/'
-  if ('input_files' in operationDefinition.input_data){
+  if ('input_files' in operationDefinition.input_data) {
     for (const image of operationDefinition.input_data.input_files) {
       image.source = 'archive'
     }
   }
 
   // first operation doesn't load unless this is here
-  function postOperationSuccess(){
+  function postOperationSuccess() {
     emit('reloadSession')
   }
-	
-  await fetchApiCall({url: url, method: 'POST', body: operationDefinition, successCallback: postOperationSuccess, failCallback: handleError})
+
+  await fetchApiCall({ url: url, method: 'POST', body: operationDefinition, successCallback: postOperationSuccess, failCallback: handleError })
 }
 
+watch(() => props.data, () => {
```

Comment:
watching objects can be expensive for vue, if you're only using the operations property here I think watching that property itself would work https://vuejs.org/guide/essentials/watchers.html#deep-watchers (second example ()=> state.someObject, state.someObject by itself won't work though it needs to be a getter which is what the anonymous function essentially is)